### PR TITLE
feat: add cursor pagination

### LINKED ISSUE #1049: Support for Cursor Pagination in nestjs-paginate
Hi, I’m wondering if nestjs-paginate currently supports cursor-based pagination in addition to the existing offset-based approach. If not, are there any plans to add this feature in the future? It’d be great for handling large datasets more efficiently. Thanks!

### PR DESCRIPTION
**Pull request by smileeunsong**
### Description
This pull request introduces support for cursor-based pagination in `nestjs-paginate`, addressing the feature request raised in [#1049](https://github.com/ppetzold/nestjs-paginate/issues/1049). Cursor pagination complements the existing offset-based approach (`LIMIT_AND_OFFSET` and `TAKE_AND_SKIP`) by providing a more efficient way to paginate large datasets, avoiding the performance issues associated with high offsets.

### Key Changes:
- **New Pagination Type**: Added `CURSOR` to the `PaginationType` enum, enabling cursor-based pagination alongside existing types.
- **Config Enhancement**: Introduced `cursorableColumns` in `PaginateConfig` to specify columns eligible for cursor pagination, ensuring flexibility and control over cursor keys.
- **Query Support**: Extended `PaginateQuery` to include `cursorColumn` and `cursorDirection` (`after` or `before`), allowing clients to define the cursor and traversal direction.
- **Response Structure**: Updated `Paginated` to include `meta.firstCursor` and `meta.lastCursor` for current page boundaries, and modified `links` to use `before` and `next` for cursor navigation (instead of `firstCursor`/`lastCursor` to align with intuitive navigation).
- **Implementation**: Modified the `paginate` function to handle cursor logic:
  - Validates `cursorableColumns`, `cursorColumn`, and `cursorDirection`.
  - Applies cursor-based filtering and sorting using TypeORM's query builder.
  - Generates navigation links with dynamic direction switching.
- **Tests**: Added comprehensive test cases in `paginate.spec.ts` to verify cursor pagination behavior, including edge cases like invalid `cursorColumn` or `cursorDirection`.

### Motivation
As noted in [#1049](https://github.com/ppetzold/nestjs-paginate/issues/1049), cursor pagination is crucial for efficiently handling large datasets, especially in scenarios where offset-based pagination becomes slow due to database index skipping. This implementation provides a robust and flexible solution while maintaining backward compatibility with existing features.

### Usage Example
```ts
const config: PaginateConfig<CatEntity> = {
    sortableColumns: ['id', 'name'],
    cursorableColumns: ['id'],
    paginationType: PaginationType.CURSOR,
    defaultLimit: 2,
};
const query: PaginateQuery = {
    path: '',
    cursorColumn: 'id',
    cursorDirection: 'after',
    cursor: '2',
};
const result = await paginate(query, catRepo, config);
```

### Notes:
- The cursor values are currently raw column values (e.g., `"2"` for `id`). Future enhancements could include cursor encryption for added security.
- The implementation avoids computing `totalItems` for cursor pagination to preserve performance benefits, aligning with common cursor-based patterns (e.g., Relay).

Closes #1049. Please review and let me know if any adjustments are needed!

### CODE REVIEW

File: src/paginate.ts [hunk 1]
```diff
@@ -220,6 +226,33 @@ export async function paginate<T extends ObjectLiteral>(
         }
     }
 
+    if (config.paginationType === PaginationType.CURSOR) {
+        const formatMessage = (msg: string) => {
+            logger.debug(msg)
+            throw new ServiceUnavailableException(msg)
+        }
+
+        if (config.cursorableColumns.length < 1) {
```
Comment (ppetzold) [inline src/paginate.ts:235]: add `if (!config.cursorableColumns ..)`

**Review by ppetzold — CHANGES_REQUESTED** [review]
well done ! add the safeguard check and a README section - and then this PR should be good to go 🙌

File: src/paginate.ts (hunk 1 above)
Comment (smileeunsong) [reply src/paginate.ts:235]:
```ts
if (!config.cursorableColumns?.length) {
    formatMessage("Missing required 'cursorableColumns' config.")
}
```
Thank you for your feedback :)
How about this one?

**Comment by smileeunsong** [comment]
### Additional Changes:
- **Date Handling**: Added logic to handle `Date` type cursor values by converting them to ISO 8601 strings (`toISOString()`) in the paginate function. This ensures consistent string representation across databases when extracting `firstCursor` and `lastCursor`.

**Comment by smileeunsong** [comment]
@ppetzold For the additional commit on handling Date, I switched to `timestamptz` for PostgreSQL to fix timezone test errors, but I’m kinda stuck on how to handle this for MariaDB and SQLite since they don’t have something like timestamptz... Any idea?

**Comment by ppetzold** [comment]
> @ppetzold For the additional commit on handling Date, I switched to `timestamptz` for PostgreSQL to fix timezone test errors, but I’m kinda stuck on how to handle this for MariaDB and SQLite since they don’t have something like timestamptz... Any idea?

umm, def remove timestamptz again and just let typeorm handle it. if the issue is cross-db consistency, maybe add an `AfterLoad()` hook to normalize `lastVetVisit` to UTC?

**Comment by smileeunsong** [comment]
> > @ppetzold For the additional commit on handling Date, I switched to `timestamptz` for PostgreSQL to fix timezone test errors, but I’m kinda stuck on how to handle this for MariaDB and SQLite since they don’t have something like timestamptz... Any idea?
> 
> umm, def remove timestamptz again and just let typeorm handle it. if the issue is cross-db consistency, maybe add an `AfterLoad()` hook to normalize `lastVetVisit` to UTC?

@ppetzold, I noticed that the internal logic was comparing values without converting them to Date objects. I managed to fix it by tweaking the logic instead of touching the database. No need for timestamptz or anything—works across DBs now. Thanks for the input!
